### ISSUE TITLE
systemd input plugin doesn't work on journald 252

### ISSUE THREAD

**Issue by devvick** [fluent/fluent-bit#6581]
## Bug Report

**Describe the bug**

systemd input plugin stopped working for me after updating to systemd 252 while using fluent-bit:1.9 docker image.
Updating image to fluent-bit:2.0 didn't help.

**To Reproduce**
- Docker command
```
docker run --rm --name fluentbit \
        --network=host \
        -v /var/log:/var/log:ro \
        -v /run/log:/run/log:ro \
        -v /run/systemd:/run/systemd:ro \
        -v /etc/machine-id:/etc/machine-id:ro \
        cr.fluentbit.io/fluent/fluent-bit:2.0 -vvv --input=systemd -p Read_From_Tail=On -p tag='*' --output=stdout
```
- Immediate log on container start
```
Fluent Bit v2.0.6
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/12/19 19:48:23] [ info] Configuration:
[2022/12/19 19:48:23] [ info]  flush time     | 1.000000 seconds
[2022/12/19 19:48:23] [ info]  grace          | 5 seconds
[2022/12/19 19:48:23] [ info]  daemon         | 0
[2022/12/19 19:48:23] [ info] ___________
[2022/12/19 19:48:23] [ info]  inputs:
[2022/12/19 19:48:23] [ info]      systemd
[2022/12/19 19:48:23] [ info] ___________
[2022/12/19 19:48:23] [ info]  filters:
[2022/12/19 19:48:23] [ info] ___________
[2022/12/19 19:48:23] [ info]  outputs:
[2022/12/19 19:48:23] [ info]      stdout.0
[2022/12/19 19:48:23] [ info] ___________
[2022/12/19 19:48:23] [ info]  collectors:
[2022/12/19 19:48:23] [ info] [fluent bit] version=2.0.6, commit=211f841fe2, pid=1
[2022/12/19 19:48:23] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2022/12/19 19:48:23] [ info] [storage] ver=1.3.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2022/12/19 19:48:23] [ info] [cmetrics] version=0.5.7
[2022/12/19 19:48:23] [ info] [ctraces ] version=0.2.5
[2022/12/19 19:48:23] [ info] [input:systemd:systemd.0] initializing
[2022/12/19 19:48:23] [ info] [input:systemd:systemd.0] storage_strategy='memory' (memory only)
[2022/12/19 19:48:23] [debug] [systemd:systemd.0] created event channels: read=21 write=22
[2022/12/19 19:48:23] [debug] [input:systemd:systemd.0] jump to the end of journal and skip 2 last entries
[2022/12/19 19:48:23] [debug] [input:systemd:systemd.0] sd_journal library may truncate values to sd_journal_get_data_threshold() bytes: 65536
[2022/12/19 19:48:23] [debug] [stdout:stdout.0] created event channels: read=26 write=29
[2022/12/19 19:48:23] [ info] [sp] stream processor started
[2022/12/19 19:48:23] [ info] [output:stdout:stdout.0] worker #0 started
```
Then, even as logs are added to the journal, they don't appear in the fluentbit output.

**Expected behavior**

Fluent-bit reads the journal from /var/log/journal, or at least produces an error.
Working example on systemd 219 using the same docker run command as above:
<details>
  <summary>
    LOG
  </summary>
<pre><code>
Fluent Bit v2.0.6
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/12/19 19:58:48] [ info] Configuration:
[2022/12/19 19:58:48] [ info]  flush time     | 1.000000 seconds
[2022/12/19 19:58:48] [ info]  grace          | 5 seconds
[2022/12/19 19:58:48] [ info]  daemon         | 0
[2022/12/19 19:58:48] [ info] ___________
[2022/12/19 19:58:48] [ info]  inputs:
[2022/12/19 19:58:48] [ info]      systemd
[2022/12/19 19:58:48] [ info] ___________
[2022/12/19 19:58:48] [ info]  filters:
[2022/12/19 19:58:48] [ info] ___________
[2022/12/19 19:58:48] [ info]  outputs:
[2022/12/19 19:58:48] [ info]      stdout.0
[2022/12/19 19:58:48] [ info] ___________
[2022/12/19 19:58:48] [ info]  collectors:
[2022/12/19 19:58:48] [ info] [fluent bit] version=2.0.6, commit=211f841fe2, pid=1
[2022/12/19 19:58:48] [debug] [engine] coroutine stack size: 196608 bytes (192.0K)
[2022/12/19 19:58:48] [ info] [storage] ver=1.3.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2022/12/19 19:58:48] [ info] [cmetrics] version=0.5.7
[2022/12/19 19:58:48] [ info] [ctraces ] version=0.2.5
[2022/12/19 19:58:48] [ info] [input:systemd:systemd.0] initializing
[2022/12/19 19:58:48] [ info] [input:systemd:systemd.0] storage_strategy='memory' (memory only)
[2022/12/19 19:58:48] [debug] [systemd:systemd.0] created event channels: read=21 write=22
[2022/12/19 19:58:48] [debug] [input:systemd:systemd.0] jump to the end of journal and skip 2 last entries
[2022/12/19 19:58:48] [debug] [input:systemd:systemd.0] sd_journal library may truncate values to sd_journal_get_data_threshold() bytes: 65536
[2022/12/19 19:58:48] [debug] [stdout:stdout.0] created event channels: read=26 write=29
[2022/12/19 19:58:48] [ info] [sp] stream processor started
[2022/12/19 19:58:48] [ info] [output:stdout:stdout.0] worker #0 started
[2022/12/19 19:58:54] [debug] [input chunk] update output instances with new chunk size diff=576
[2022/12/19 19:58:54] [debug] [input chunk] update output instances with new chunk size diff=621
[2022/12/19 19:58:54] [debug] [input chunk] update output instances with new chunk size diff=627
[2022/12/19 19:58:54] [debug] [input chunk] update output instances with new chunk size diff=723
[2022/12/19 19:58:54] [debug] [input chunk] update output instances with new chunk size diff=653
[2022/12/19 19:58:55] [debug] [task] created task=0xffff80010090 id=0 OK
[2022/12/19 19:58:55] [debug] [task] created task=0xffff80010170 id=1 OK
[2022/12/19 19:58:55] [debug] [task] created task=0xffff80010260 id=2 OK
[2022/12/19 19:58:55] [debug] [task] created task=0xffff80010360 id=3 OK
[2022/12/19 19:58:55] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2022/12/19 19:58:55] [debug] [output:stdout:stdout.0] task_id=1 assigned to thread #0
[2022/12/19 19:58:55] [debug] [output:stdout:stdout.0] task_id=2 assigned to thread #0
[2022/12/19 19:58:55] [debug] [output:stdout:stdout.0] task_id=3 assigned to thread #0
[2022/12/19 19:58:55] [debug] [out flush] cb_destroy coro_id=0
[2022/12/19 19:58:55] [debug] [out flush] cb_destroy coro_id=1
[2022/12/19 19:58:55] [debug] [out flush] cb_destroy coro_id=2
[2022/12/19 19:58:55] [debug] [task] destroy task=0xffff80010090 (task_id=0)
[2022/12/19 19:58:55] [debug] [out flush] cb_destroy coro_id=3
[2022/12/19 19:58:55] [debug] [task] destroy task=0xffff80010170 (task_id=1)
[2022/12/19 19:58:55] [debug] [task] destroy task=0xffff80010260 (task_id=2)
[2022/12/19 19:58:55] [debug] [task] destroy task=0xffff80010360 (task_id=3)
[0] sshd.service: [1671479934.243993000, {"PRIORITY"=>"6", "_UID"=>"0", "_GID"=>"0", "_CAP_EFFECTIVE"=>"1ffffffffff"
(...)
</code></pre>
</details>

**Your Environment**
* Version used:
**2.0.6**
* Configuration: 
**CLI-only**
* Environment name and version (e.g. Kubernetes? What version?):
**Docker 20.10.21**
* Server type and version:
**AWS EC2**
* Operating System and version:
  * Arch Linux
  * **systemd 252 (252.3-1-arch)**
+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified
* Filters and plugins:
**none**

**Comment by nokute78** [comment]
It may be a library issue.
A version of libsystemd in the fluent-bit container is less than 252 and it may not parse a journal that is created by systemd v252.

I tested following steps.
1. Copy from ArchLinux(systemd v252) journal to Ubuntu 22.04(systemd v251).
2. Compile following test code on Ubuntu. e.g. `gcc -lsystemd systemd.c`
3. Exec program. It needs to pass a directory of journal.
4. It can't parse journal of ArchLinux. On the otherhand it can parse journal of Ubuntu.

```c
/* SPDX-License-Identifier: MIT-0 */
/* Based https://www.freedesktop.org/software/systemd/man/sd_journal_next.html */
#include <errno.h>
#include <stdio.h>
#include <systemd/sd-journal.h>

int main(int argc, char *argv[]) {
  int r;
  sd_journal *j;

  if (argc < 2) {
    fprintf(stderr, "arg error\n");
    return 1;
  }

  r = sd_journal_open_directory(&j, argv[1], SD_JOURNAL_SYSTEM);
  if (r < 0) {
    errno = -r;
    fprintf(stderr, "Failed to open journal: %m\n");
    return 1;
  }
  SD_JOURNAL_FOREACH(j) {
    const char *d;
    size_t l;

    r = sd_journal_get_data(j, "MESSAGE", (const void **)&d, &l);
    if (r < 0) {
      errno = -r;
      fprintf(stderr, "Failed to read message field: %m\n");
      continue;
    }

    printf("%.*s\n", (int) l, d);
  }
  sd_journal_close(j);
  return 0;
}
```

**Comment by chitoku-k** [comment]
It is now fixed in [v2.0.9](https://github.com/fluent/fluent-bit/tree/v2.0.9) as it includes [libsystemd0](https://packages.debian.org/en/bullseye-backports/libsystemd0) 252.4-1~bpo11+1.
